### DISCUSSION TITLE
Project Feedback

### DISCUSSION THREAD

**Discussion by darkstego** [darkstego/Mudeer#11] [General]
Any feedback or review is welcome. I am interested in knowing what are things you like or don't like? Are there special ways you use the project?

**Comment by Boris-Barboris** [comment]
Good job, I liked the hotkey convention!

I will reiterate the complementary request to the https://github.com/darkstego/Mudeer/issues/12:
For me, 21:9 user, there is a need for a special, non-tilable script that will place the floatable window as if it's filling a 16:9 screen, dead-center.

For 21:9 monitor the arrangement would be [2.5 gap, 16 window, 2.5 gap]
For 32:9 monitor the arrangement would be [8 gap, 16 window, 8 gap]

**Reply by darkstego** [reply]
So listening to the feedback here and in #12 it seems some users are looking for a center floating window. I am not too familiar with this use-case, so correct me if I am wrong, but this basically provides an alternative to the maximizing windows, which might not be practical in very wide monitors. I would assume that the small gaps on either side, especially on a 21:9 monitor, would be unused. 

Based on my understanding, I am thinking of a center float hotkey. The width of the which can be customized in the config GUI. Would that work well?

**Reply by Boris-Barboris** [reply]
> I would assume that the small gaps on either side, especially on a 21:9 monitor, would be unused.   

Yep, unused or overlapped, it's not a problem sometimes.

> Center floating window.   

I guess, if you define floating as a window that can overlap with other, tiled windows.

**Reply by darkstego** [reply]
> I guess, if you define floating as a window that can overlap with other, tiled windows.

The script does soft tiling, so it always allows overlap.

I will look into setting a center float option.

**Reply by Boris-Barboris** [reply]
If you didn't see it already, I am perfectly comfortable with https://github.com/darkstego/Mudeer/pull/13 on 21:9

**Comment by matthews-jca** [comment]
My major use case is a replacement for the fancy-zones windows powertool along with shortcuts for using a 1/4-1/2-1/4 window layout and streamdeck to auto align windows for focus.

32:9 monitor, typically want windows in [work chat][current productivity focus][email/calendar/reference docs]
Productivity focus can be a single document in browser or 5+ windows of terminals, output, etc

The biggest feature to me was to setup the layout I wanted then give the shortcuts to https://timothycrosley.github.io/streamdeck-ui/ as a macropad so when new windows are launched or focus changes I can give focus and move to my active area quickly.

The project has been working quite well for all my needs

**Reply by darkstego** [reply]
Just a question, when you have 5+ windows in the productivity zone, what do you use to organizing them? Are you still tiling them or are they one program?

**Reply by matthews-jca** [reply]
That's a good question. At this point usually it's overlapping, spilling over into other zones, and otherwise messy. I never even though about how I could be more effective at arranging them. 

We are working on ROS robotics applications, so it's usually a combination of browser windows, terminal output, and robot GUIs where you typically want to see as many things at once

**Comment by algo124** [comment]
Hello! I love using Mudeer. Do you know if it'll work on Plasma 6? If not, do you figure you'll update it? Thanks so much.

**Reply by darkstego** [reply]
It does work on Plasma 6, but you need to use the kde6 branch, and manually install it. The main branch is plasma 5 only for now, once I migrate over I will switch main branch to plasma 6.

**Reply by sulaweyo** [reply]
I have a related question to Plasma 6. I  have switched to the branch but for the best of me I fail to get global shortcuts to work again (Arch, X11 or Wayland). I use Mudeer shortcuts mostly to move terminal windows on my widescreen but since the upgrade I struggle to get anything working again. The shortcuts are not caught before they reach the terminal and just end up as inputs there. I'm most likely missing something totally obvious so any hint would be highly appreciated :)

**Reply by darkstego** [reply]
I added a section to the README. Under trouble shooting to fix non-working shortcuts.

**Reply by sulaweyo** [reply]
Thank you for your comments. Turned out my problem was caused by having the old version still present next to the plasma6 one. I cleaned up .local/share/kwin/scripts and .config/kglobalshortcutsrc as well as .config/kwinrc then reinstalled and everything started working as expected again. Should have been the first thing to do...
Thank you again for Mudeer @darkstego

**Comment by stoper123** [comment]
can you make quick tutorial  video on how to install it

**Reply by stoper123** [reply]
C:\Users\parth>git clone https://github.com/darkstego/Mudeer.git
fatal: destination path 'Mudeer' already exists and is not an empty directory.

C:\Users\parth>cd Mudeer && make install
'make' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\parth\Mudeer>make install
'make' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\parth\Mudeer>install
'install' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\parth\Mudeer>

**Comment by stoper123** [comment]
is it for window or linux

**Reply by darkstego** [reply]
Linux KDE Only

**Comment by plasmobit** [comment]
Thank you a lot, very useful extension! :+1:

**Comment by UpscalePilot** [comment]
I love this project, and I've been utterly heartborken this afternoon to find out it doesn't work on KDE Plasma 6 after I did a system update. Please, please, please consider creating an updated version that works with Plasma 6

**Reply by darkstego** [reply]
There is a Plasma 6 Version, on the Plasma 6 branch. It will be mainlined in a day or two.

**Reply by UpscalePilot** [reply]
Oh, I didn't see that! Thank you

**Comment by TraceyC77** [comment]
I've been missing this plugin since my system was updated to Plasma 6, and I'm so happy to see it has been updated. Thanks for your work on this.

I ran into something that was unclear when I went to reinstall the plugin. In the "Pick Your Installation Option" window, the differences between the three options is not clear.

mudeer-20230619013108.kwinscript
mudeer.5.0.kwinscript
mudeer.kwinscript

I assume the middle one is the one I want. It would be better if there was a way to provide descriptions or make it more clear via the names what the difference between these options was, and when a user might want one over the others.

**Reply by darkstego** [reply]
I didn't know how it appears. I will clear that up.

**Comment by Jarco** [comment]
Thank you so much for making this Plasma 6 compatible! I was so lost without this script.

**Comment by bashnetfr** [comment]
YOOOO! You the GOAT man.
